cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetServiceAuthorizationTest.java

```diff
@@ -168,20 +173,33 @@ public void execute() throws Exception {
       }
     }, String.format("Alice should not be able to delete instance %s since she does not have privileges", dsId1));
     grantAndAssertSuccess(dsId1, ALICE, ImmutableSet.of(Action.ADMIN));
-    Assert.assertEquals(ImmutableSet.of(dsId1, dsId2),
+    Assert.assertEquals(ImmutableSet.of(dsId1, dsId2, dsId),
                         summaryToDatasetIdSet(dsFramework.getInstances(NamespaceId.DEFAULT)));
     // since Alice now is ADMIN for dsId1, she should be able to delete it
     dsFramework.deleteInstance(dsId1);
 
-    // Now Alice only see dsId2 from list.
-    Assert.assertEquals(ImmutableSet.of(dsId2),
+    // Now Alice only see dsId2 and dsId from list.
+    Assert.assertEquals(ImmutableSet.of(dsId2, dsId),
                         summaryToDatasetIdSet(dsFramework.getInstances(NamespaceId.DEFAULT)));
 
-    // Bob should still see 1 instance
+    // Bob should be able to see dsId and dsId2
     SecurityRequestContext.setUserId(BOB.getName());
-    Assert.assertEquals(ImmutableSet.of(dsId2),
+    Assert.assertEquals(ImmutableSet.of(dsId2, dsId),
                         summaryToDatasetIdSet(dsFramework.getInstances(NamespaceId.DEFAULT)));
     dsFramework.deleteInstance(dsId2);
+
+    SecurityRequestContext.setUserId(ALICE.getName());
+    dsFramework.deleteInstance(dsId);
+
+    grantAndAssertSuccess(dsId2, ALICE, EnumSet.of(Action.ADMIN));
+    // add add the instance again
+    dsFramework.addInstance(Table.class.getName(), dsId, DatasetProperties.EMPTY);
+    dsFramework.addInstance(Table.class.getName(), dsId1, DatasetProperties.EMPTY);
+    dsFramework.addInstance(Table.class.getName(), dsId2, DatasetProperties.EMPTY);
+
+    // should be successful since ALICE has ADMIN on all datasets
+    dsFramework.deleteAllInstances(NamespaceId.DEFAULT);
```

Comment:
Before `deleteAllInstances()` let's list the instances and make sure there are three of them.